pkgs/development/compilers/llvm/common/libc/default.nix

```
@@ -46,6 +46,7 @@ stdenv.mkDerivation (finalAttrs: {
   outputs = [ "out" ] ++ (lib.optional isFullBuild "dev");
 
   postUnpack = lib.optionalString isFullBuild ''
+    chmod u+rw $sourceRoot/../$pname
```

This non-recursive chmod does not fix the build, because it does not affect the directory where patchShebangs is trying to add a temporary file.  Perhaps you want this?

```suggestion
    chmod +w $sourceRoot/../$pname/utils/hdrgen
```

Remember to test your changes!

---

Huh wait why did it work for me then?

---

Oh, I accidentally did an overlay build. I've fixed full now.